tests/test_bootstrap.py

```diff
@@ -35,6 +36,13 @@
 VERSION_PATH = os.path.join(get_test_config_dir(), config_util.VERSION_FILE)
 
 
+@pytest.fixture(autouse=True)
```

Comment:
Since we are into incremental gains here, might there be a further improvement by setting scope='session' on these autouse fixtures?

Reply:
I think we would break the requirements tests if we did it elsewhere because they rely on it not being patched. We would have to undo the patching for the requirements tests which is a bit more involved.  Something to think about for a future PR though

Reply:
I think the scope parameter just defines how far the fixture is used before it reset. 

As fast as I know it shouldn't make it run in any more places than it does already.

Reply:
If we change the scope to session any test that runs after it will have the patch set. We have had some problems with that before. See #116195

> session: the fixture is destroyed at the end of the test session.
https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session